src/bugreporter/odemis_bugreporter.py

```diff
@@ -0,0 +1,540 @@
+        else:
+            # just pick every potential microscope model
+            models = glob(os.path.join(odemis_config['CONFIGPATH'], '*/*.odm.yaml'))
+        files.extend(models)
+
+        # Add the latest overlay-report if it's possibly related (ie, less than a day old)
+        overlay_reps = glob(os.path.join(home_dir, 'odemis-overlay-report/*'))
+        if overlay_reps and (time.time() - os.path.getmtime(overlay_reps[-1])) / 3600 < 24:
+            files.append(overlay_reps[-1])
+
+        # Add the latest DELPHI calibration report if it's possibly related (ie, less than a day old)
+        delphi_calib_reps = glob(os.path.join(home_dir, 'delphi-calibration-report/*'))
+        if delphi_calib_reps and (time.time() - os.path.getmtime(delphi_calib_reps[-1])) / 3600 < 24:
+            files.append(delphi_calib_reps[-1])
+
+        # Save hw status (if available)
+        ret_code = subprocess.call(['odemis-cli', '--check'])
+        if ret_code in (0, 3):
+            try:
+                # subprocess doesn't have timeout argument in python 2.x, so use future instead
+                executor = futures.ThreadPoolExecutor(max_workers=1)
+                f = executor.submit(subprocess.check_output, ['odemis-cli', '--list-prop', '*'])
+                props = f.result(60)
+                hwfn = "/tmp/odemis-hw-status.txt"
+                with open(hwfn, 'w+') as f:
+                    f.write(props)
+                files.append(hwfn)
+            except:
+                logging.warning("Cannot save hw status.")
+
+        # Compress files
+        with zipfile.ZipFile(self.zip_fn, "w") as archive:
+            for f in files:
+                if os.path.isfile(f):
+                    archive.write(f, compress_type=zipfile.ZIP_DEFLATED)
+                else:
+                    logging.warning("Bugreporter could not find file %s." % f)
+
+    def set_description(self, name, email, subject, message):
+        """
+        Saves the description parameters for the ticket creation. Sets description_done
+        flag to True. Writes arguments to tmp file.
+        :arg name, email, summary, description: (String) arguments for corresponding dictionary
+        keys
+        :modifies self.report_description: (String --> String)
+        """
+        self.report_description = {'name': name,
+                                   'email': email,
+                                   'subject': subject,
+                                   'message': message}
+        # Create ticket with special id when testing
+        if TEST_SUPPORT_TICKET:
+            self.report_description['topicId'] = 12
+
+        with open('/tmp/description.txt', 'a') as f:
+            f.write('Name: %s\n' % name)
+            f.write('Email: %s\n' % email)
+            f.write('Summary: %s\n\n' % subject)
+            f.write('Description:\n%s' % message)
+
+        self._compress_files_f.add_done_callback(self._on_compression_done)
```

Comment:
That probably doesn't work so well, because it will likely run in the current thread if the future is already finished.
I'd suggest to have a new function call `send_report(name, email, subject, message)`, which just put a call the _set_description in a future (so running in a separate thread).
Then, `_set_description()` can start by `self._compress_files_f.result()`, and then continue with what is currently in `_on_compression_done()`.